manifests/init.pp

```diff
@@ -12,155 +12,65 @@
-  $pam_debug_state                      = $::winbind::params::pam_debug_state,
-  $pam_debug                            = $::winbind::params::pam_debug,
-  $pam_krb5_auth                        = $::winbind::params::pam_krb5_auth,
-  $pam_krb5_ccache_type                 = $::winbind::params::pam_krb5_ccache_type,
-  $pam_mkhomedir                        = $::winbind::params::pam_mkhomedir,
-  $pam_require_membership_of            = $::winbind::params::pam_require_membership_of,
-  $pam_silent                           = $::winbind::params::pam_silent,
-  $pam_warn_pwd_expire                  = $::winbind::params::pam_warn_pwd_expire,
-  $smb_encrypt_passwords                = $::winbind::params::smb_encrypt_passwords,
-  $smb_idmap_config_default_backend     = $::winbind::params::smb_idmap_config_default_backend,
-  $smb_idmap_config_default_range_end   = $::winbind::params::smb_idmap_config_default_range_end,
-  $smb_idmap_config_default_rangesize   = $::winbind::params::smb_idmap_config_default_rangesize,
-  $smb_idmap_config_default_range_start = $::winbind::params::smb_idmap_config_default_range_start,
-  $smb_include_dir                      = $::winbind::params::smb_includes_dir,
-  $smb_includes_files                   = $::winbind::params::smb_includes_files,
-  $smb_log_file                         = $::winbind::params::smb_log_file,
-  $smb_log_level                        = $::winbind::params::smb_log_level,
-  $smb_max_log_size                     = $::winbind::params::smb_max_log_size,
-  $smb_printcap_name                    = $::winbind::params::smb_printcap_name,
-  $smb_printing                         = $::winbind::params::smb_printing,
-  $smb_realm                            = $::winbind::params::smb_realm,
-  $smb_security                         = $::winbind::params::smb_security,
-  $smb_server_string                    = $::winbind::params::smb_server_string,
-  $smb_settings_hash                    = $::winbind::params::smb_settings_hash,
-  $smb_syslog                           = $::winbind::params::smb_syslog,
-  $smb_template_homedir                 = $::winbind::params::smb_template_homedir,
-  $smb_template_shell                   = $::winbind::params::smb_template_shell,
-  $smb_winbind_enum_groups              = $::winbind::params::smb_winbind_enum_groups,
-  $smb_winbind_enum_users               = $::winbind::params::smb_winbind_enum_users,
-  $smb_winbind_normalize_names          = $::winbind::params::smb_winbind_normalize_names,
-  $smb_winbind_nss_info                 = $::winbind::params::smb_winbind_nss_info,
-  $smb_winbind_offline_logon            = $::winbind::params::smb_winbind_offline_logon,
-  $smb_winbind_separator                = $::winbind::params::smb_winbind_separator,
-  $smb_winbind_use_default_domain       = $::winbind::params::smb_winbind_use_default_domain,
-  $smb_workgroup                        = $::winbind::params::smb_workgroup,
+  $enable_sharing,
```

Comment:
1. They dont have types yet because my first step was just to convert all existing code, so the types that are there now were just converted from the deprecated `validate_` functions from stdlib. Will definitely give them all types before completing.

1. I can certainly align them in columns if you like. I was just following the [guide](https://docs.puppet.com/puppet/5.0/bgtm.html#class-design), which does not have them in columns for some reason.